src/main/java/kakao/rebit/auth/service/KakaoAuthService.java

```diff
@@ -0,0 +1,68 @@
+package kakao.rebit.auth.service;
+
+import jakarta.transaction.Transactional;
+import kakao.rebit.auth.token.AuthTokenGenerator;
+import kakao.rebit.auth.dto.KakaoUserInfo;
+import kakao.rebit.auth.dto.LoginResponse;
+import kakao.rebit.auth.token.AuthToken;
+import kakao.rebit.member.entity.Member;
+import kakao.rebit.member.entity.Role;
+import kakao.rebit.member.repository.MemberRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class KakaoAuthService {
+
+    private final KakaoApiClient kakaoApiClient;
+    private final MemberRepository memberRepository;
+    private final AuthTokenGenerator authTokensGenerator;
+
+    @Autowired
+    public KakaoAuthService(KakaoApiClient kakaoApiClient, MemberRepository memberRepository,
+        AuthTokenGenerator authTokensGenerator) {
+        this.kakaoApiClient = kakaoApiClient;
+        this.memberRepository = memberRepository;
+        this.authTokensGenerator = authTokensGenerator;
+    }
+
+    public LoginResponse kakaoLogin(String code) {
+        // 1. 카카오로부터 액세스 토큰을 받음
+        String accessToken = kakaoApiClient.getAccessToken(code);
+
+        // 2. 받은 액세스 토큰으로 유저 정보를 가져옴
+        KakaoUserInfo userInfo = kakaoApiClient.getUserInfo(accessToken);
+
+        // 3. 회원 조회 또는 생성 메서드 호출
+        Member member = findOrCreateMemberByEmail(userInfo, accessToken);
+
+        // 4. JWT 토큰 생성
+        AuthToken tokens = generateAuthToken(member);
+
+        return new LoginResponse(tokens);
+    }
+
+    @Transactional
+    public Member findOrCreateMemberByEmail(KakaoUserInfo userInfo, String accessToken) {
+        String email = userInfo.kakaoAccount().email();
+
+        // 이메일로 회원을 조회하고 없으면 새로운 회원 생성
+        return memberRepository.findByEmail(email)
+            .orElseGet(() -> newMember(userInfo, accessToken, email));
+    }
+
+    private Member newMember(KakaoUserInfo userInfo, String accessToken, String email) {
+        // 사용자 정보를 바탕으로 새로운 회원 생성
+        String nickname = userInfo.properties().nickname();
+        Member newMember = new Member(nickname, "", "", email, Role.ROLE_USER, 0, accessToken);
```

Comment:
```suggestion
        Member newMember = Member.init(nickname, email, accessToken);
```
이렇게 정적 메서드를 사용하면 메서드 명을 통해 어떤 객체가 생성되는지 의미를 잘 전달해줄 수 있어요. 그리고  하드코딩되어 있는 값(초기세팅값) 들을 정적 메서드로 옮겨 객체 생성시 잘못된 값을 전달하는 것을 방지할 수 있어요. 참고해주세요~ 꼭 변경할 필요는 없습니다!